// @rpldy/shared-ui/lib/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "NoDomUploady", {
  enumerable: true,
  get: function () {
    return _NoDomUploady.default;
  }
});
Object.defineProperty(exports, "UploadyContext", {
  enumerable: true,
  get: function () {
    return _UploadyContext.default;
  }
});
Object.defineProperty(exports, "assertContext", {
  enumerable: true,
  get: function () {
    return _assertContext.default;
  }
});
Object.defineProperty(exports, "createContextApi", {
  enumerable: true,
  get: function () {
    return _UploadyContext.createContextApi;
  }
});
Object.defineProperty(exports, "generateUploaderEventHook", {
  enumerable: true,
  get: function () {
    return _hooksUtils.generateUploaderEventHook;
  }
});
Object.defineProperty(exports, "generateUploaderEventHookWithState", {
  enumerable: true,
  get: function () {
    return _hooksUtils.generateUploaderEventHookWithState;
  }
});
Object.defineProperty(exports, "getIsUploadOptionsComponent", {
  enumerable: true,
  get: function () {
    return _utils.getIsUploadOptionsComponent;
  }
});
Object.defineProperty(exports, "getUploadyVersion", {
  enumerable: true,
  get: function () {
    return _uploadyVersion.getVersion;
  }
});
Object.defineProperty(exports, "logWarning", {
  enumerable: true,
  get: function () {
    return _utils.logWarning;
  }
});
Object.defineProperty(exports, "markAsUploadOptionsComponent", {
  enumerable: true,
  get: function () {
    return _utils.markAsUploadOptionsComponent;
  }
});
Object.defineProperty(exports, "useAbortAll", {
  enumerable: true,
  get: function () {
    return _useAbortAll.default;
  }
});
Object.defineProperty(exports, "useAbortBatch", {
  enumerable: true,
  get: function () {
    return _useAbortBatch.default;
  }
});
Object.defineProperty(exports, "useAbortItem", {
  enumerable: true,
  get: function () {
    return _useAbortItem.default;
  }
});
Object.defineProperty(exports, "useAllAbortListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useAllAbortListener;
  }
});
Object.defineProperty(exports, "useBatchAbortListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchAbortListener;
  }
});
Object.defineProperty(exports, "useBatchAddListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchAddListener;
  }
});
Object.defineProperty(exports, "useBatchCancelledListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchCancelledListener;
  }
});
Object.defineProperty(exports, "useBatchErrorListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchErrorListener;
  }
});
Object.defineProperty(exports, "useBatchFinalizeListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchFinalizeListener;
  }
});
Object.defineProperty(exports, "useBatchFinishListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchFinishListener;
  }
});
Object.defineProperty(exports, "useBatchProgressListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchProgressListener;
  }
});
Object.defineProperty(exports, "useBatchStartListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useBatchStartListener;
  }
});
Object.defineProperty(exports, "useItemAbortListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemAbortListener;
  }
});
Object.defineProperty(exports, "useItemCancelListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemCancelListener;
  }
});
Object.defineProperty(exports, "useItemErrorListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemErrorListener;
  }
});
Object.defineProperty(exports, "useItemFinalizeListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemFinalizeListener;
  }
});
Object.defineProperty(exports, "useItemFinishListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemFinishListener;
  }
});
Object.defineProperty(exports, "useItemProgressListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemProgressListener;
  }
});
Object.defineProperty(exports, "useItemStartListener", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useItemStartListener;
  }
});
Object.defineProperty(exports, "useRequestPreSend", {
  enumerable: true,
  get: function () {
    return _eventListenerHooks.useRequestPreSend;
  }
});
Object.defineProperty(exports, "useUploadOptions", {
  enumerable: true,
  get: function () {
    return _useUploadOptions.default;
  }
});
Object.defineProperty(exports, "useUploady", {
  enumerable: true,
  get: function () {
    return _useUploadyContext.default;
  }
});
Object.defineProperty(exports, "useUploadyContext", {
  enumerable: true,
  get: function () {
    return _useUploadyContext.default;
  }
});
Object.defineProperty(exports, "withBatchStartUpdate", {
  enumerable: true,
  get: function () {
    return _withBatchStartUpdate.default;
  }
});
Object.defineProperty(exports, "withRequestPreSendUpdate", {
  enumerable: true,
  get: function () {
    return _withRequestPreSendUpdate.default;
  }
});

var _eventListenerHooks = require("./hooks/eventListenerHooks");

var _utils = require("./utils");

var _hooksUtils = require("./hooks/hooksUtils");

var _uploadyVersion = require("./uploadyVersion");

var _NoDomUploady = _interopRequireDefault(require("./NoDomUploady"));

var _assertContext = _interopRequireDefault(require("./assertContext"));

var _UploadyContext = _interopRequireWildcard(require("./UploadyContext"));

var _useUploadOptions = _interopRequireDefault(require("./hooks/useUploadOptions"));

var _useAbortItem = _interopRequireDefault(require("./hooks/useAbortItem"));

var _useAbortBatch = _interopRequireDefault(require("./hooks/useAbortBatch"));

var _useAbortAll = _interopRequireDefault(require("./hooks/useAbortAll"));

var _useUploadyContext = _interopRequireDefault(require("./hooks/useUploadyContext"));

var _withRequestPreSendUpdate = _interopRequireDefault(require("./hocs/withRequestPreSendUpdate"));

var _withBatchStartUpdate = _interopRequireDefault(require("./hocs/withBatchStartUpdate"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
