// @rpldy/uploader/lib/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  createUploader: true,
  composeEnhancers: true,
  UPLOADER_EVENTS: true,
  DEFAULT_OPTIONS: true,
  FILE_STATES: true,
  BATCH_STATES: true
};
Object.defineProperty(exports, "BATCH_STATES", {
  enumerable: true,
  get: function () {
    return _shared.BATCH_STATES;
  }
});
Object.defineProperty(exports, "DEFAULT_OPTIONS", {
  enumerable: true,
  get: function () {
    return _defaults.DEFAULT_OPTIONS;
  }
});
Object.defineProperty(exports, "FILE_STATES", {
  enumerable: true,
  get: function () {
    return _shared.FILE_STATES;
  }
});
Object.defineProperty(exports, "UPLOADER_EVENTS", {
  enumerable: true,
  get: function () {
    return _consts.UPLOADER_EVENTS;
  }
});
Object.defineProperty(exports, "composeEnhancers", {
  enumerable: true,
  get: function () {
    return _composeEnhancers.default;
  }
});
Object.defineProperty(exports, "createUploader", {
  enumerable: true,
  get: function () {
    return _uploader.default;
  }
});
exports.default = void 0;

var _uploader = _interopRequireDefault(require("./uploader"));

var _composeEnhancers = _interopRequireDefault(require("./composeEnhancers"));

var _consts = require("./consts");

var _defaults = require("./defaults");

var _sender = require("@rpldy/sender");

Object.keys(_sender).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _sender[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _sender[key];
    }
  });
});

var _shared = require("@rpldy/shared");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = _uploader.default;
exports.default = _default;
